rinja_parser/src/expr.rs

```diff
@@ -188,7 +190,56 @@ impl<'a> Expr<'a> {
     expr_prec_layer!(band, shifts, token_bitand);
     expr_prec_layer!(shifts, addsub, alt((tag(">>"), tag("<<"))));
     expr_prec_layer!(addsub, muldivmod, alt((tag("+"), tag("-"))));
-    expr_prec_layer!(muldivmod, filtered, alt((tag("*"), tag("/"), tag("%"))));
+    expr_prec_layer!(muldivmod, is_defined, alt((tag("*"), tag("/"), tag("%"))));
+
+    fn is_defined(i: &'a str, level: Level) -> ParseResult<'a, WithSpan<'a, Self>> {
+        let (_, level) = level.nest(i)?;
+        let start = i;
+        let (i, lhs) = Self::filtered(i, level)?;
+        let (i, rhs) = opt(preceded(
+            ws(keyword("is")),
+            opt(terminated(opt(keyword("not")), ws(keyword("defined")))),
+        ))(i)?;
+        let is_neg = match rhs {
+            None => return Ok((i, lhs)),
+            Some(None) => {
+                return Err(nom::Err::Failure(ErrorContext::new(
+                    "expected `defined` or `not defined` after `is`",
+                    // We use `start` to show the whole `var is` thing instead of the current token.
+                    start,
+                )));
+            }
+            Some(Some(None)) => false,
+            Some(Some(Some(_))) => true,
+        };
+        match *lhs {
+            Self::Var(_) => {}
```

Comment:
Please capture the variable name in here.

Reply:
Done!